src/wallet/asyncrpcoperation_common.cpp

```diff
@@ -40,3 +41,124 @@ std::pair<CTransaction, UniValue> SignSendRawTransaction(UniValue obj, std::opti
 
     return std::make_pair(tx, sendResult);
 }
+
+void ThrowInputSelectionError(
+        const InputSelectionError& err,
+        const ZTXOSelector& selector,
+        const TransactionStrategy& strategy)
+{
+    std::visit(match {
+        [](const AddressResolutionError& err) {
+            switch (err) {
+                case AddressResolutionError::SproutRecipientsNotSupported:
+                    throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        "Sending funds into the Sprout pool is no longer supported.");
+                case AddressResolutionError::TransparentRecipientNotAllowed:
+                    throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        "This transaction would have transparent recipients, which is not "
+                        "enabled by default because it will publicly reveal transaction "
+                        "recipients and amounts. THIS MAY AFFECT YOUR PRIVACY. Resubmit "
+                        "with the `privacyPolicy` parameter set to `AllowRevealedRecipients` "
+                        "or weaker if you wish to allow this transaction to proceed anyway.");
+                case AddressResolutionError::RevealingSaplingAmountNotAllowed:
+                    throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        "Could not send to the Sapling shielded pool without spending non-Sapling "
+                        "funds, which would reveal transaction amounts. THIS MAY AFFECT YOUR "
+                        "PRIVACY. Resubmit with the `privacyPolicy` parameter set to "
+                        "`AllowRevealedAmounts` or weaker if you wish to allow this transaction to "
+                        "proceed anyway.");
+                case AddressResolutionError::CouldNotResolveReceiver:
+                    throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        "Could not send to an Orchard-only receiver, despite a lax privacy policy. "
+                        "Either there are insufficent non-Sprout funds, or NU5 has not been "
```

Comment:
```suggestion
                        "Either there are insufficient non-Sprout funds, or NU5 has not been "
```

Reply:
It might not be clear why non-Sprout funds are needed. (It is because we cannot send directly from Sprout to Orchard because no transaction version supports both, but that's not obvious.)